Documentation/DesignGuidelines.md

```diff
@@ -6,28 +6,459 @@ Guidelines](http://blogs.msdn.com/b/rxteam/archive/2010/10/28/rx-design-guidelin
+them along the way. In other words, most [operators][] immediately stop doing work
+when an error is received, and then propagate the error onward. This even
+applies to time-shifted operators, like [`delay`][delay]—which, despite its name, will
+forward any errors immediately.
+
+Consequently, errors should only be used to represent “abnormal” termination. If
+it is important to let operators (or consumers) finish their work, a `Next`
+event describing the result might be more appropriate.
+
+#### Completion indicates success
+
+An event stream sends `Completed` when the operation has completed successfully,
+or to indicate that the stream has terminated normally.
+
+Many operators manipulate the `Completed` event to shorten or extend the
+lifetime of an event stream.
+
+For example, [`take`][take] will complete after the specified number of values have
+been received, thereby terminating the stream early. On the other hand, most
+operators that accept multiple signals or producers will wait until _all_ of
+them have completed before forwarding a `Completed` event, since a successful
+outcome will usually depend on all the inputs.
+
+#### Interruption cancels outstanding work and usually propagates immediately
+
+An `Interrupted` event is sent when an event stream should cancel processing.
+Interruption is somewhere between [success](#completion-indicates-success)
+and [failure](#errors-behave-like-exceptions-and-propagate-immediately)—the
+operation was not successful, because it did not get to finish, but it didn’t
+necessarily “fail” either.
+
+Most [operators][] will propagate interruption immediately, but there are some
+exceptions. For example, the [flattening operators][flatten] will ignore
+`Interrupted` events that occur on the _inner_ producers, since the cancellation
+of an inner operation should not necessarily cancel the larger unit of work.
+
+RAC will automatically send an `Interrupted` event upon [disposal][Disposables], but it can
+also be sent manually if necessary. Additionally, [custom
+operators](#implementing-new-operators) must make sure to forward interruption
+events to the observer.
+
+#### Events are serial
+
+RAC guarantees that all events upon a stream will arrive serially. In other
+words, it’s impossible for the observer of a signal or producer to receive
+multiple `Event`s concurrently, even if the events are sent on multiple threads
+simultaneously.
+
+This simplifies [operator][Operators] implementations and [observers][].
+
+#### Events cannot be sent recursively
+
+Just like RAC guarantees that [events will not be received
+concurrently](#events-are-serial), it also guarantees that they won’t be
+received recursively. As a consequence, [operators][] and [observers][] _do not_ need to
+be reentrant.
+
+If an event is sent upon a signal from a thread that is _already processing_
+a previous event from that signal, deadlock will result. This is because
```

Comment:
I meant to write it like this. By way of comparison, “if XYZ, serious injury may result.”